tools/generate-helpers/pg-table-bindings/store_test.go.tpl

```diff
@@ -74,9 +74,26 @@ func (s *{{$namePrefix}}StoreSuite) TestStore() {
 	s.Nil(found{{.TrimmedType|upperCamelCase}})
 
     {{if not .JoinTable -}}
-    {{- if or (eq .ResourceType "globallyScoped") (eq .ResourceType "permissionChecker") }}
+    {{- if or (.Obj.IsGloballyScoped) (.Obj.HasPermissionChecker) (.Obj.IsDirectlyScoped)}}
     withNoAccessCtx := sac.WithNoAccess(ctx)
     {{- end }}
+	{{- if .Obj.IsDirectlyScoped }}
+	ctxWithAccessToDifferentNs := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(targetResource),
+			sac.ClusterScopeKeys({{ $name | .Obj.GetClusterID }}),
+			sac.NamespaceScopeKeys("unknown ns")))
```

Comment:
The test logic here is valid for both namespace and cluster scope, but not for the same reasons.
For namespaces, the object is not in scope because the object namespace is different from the allowed namespace.
For clusters, the object is not in scope because the object cluster is only partially allowed.
Would it make sense to extensively cover the possible invalid scopes (wrong access level, wrong cluster, partial cluster with right namespace, partial cluster with wrong namespace ...) ? Extensive coverage might require a dedicated file as the number of cases can be quite high.

Reply:
How about adding more tests in a separated PR as we have a separated task for it [ROX-10072](https://issues.redhat.com/browse/ROX-10072)

Reply:
I added tests but maybe we can reorganize test to have smaller scope and better error messages. Now we have a single full workflow test I think it make sens to have a separated suite just for sac with all cases covered.